### ISSUE TITLE
Raising PauliSum to integer power gives incorrect results

### ISSUE THREAD

**Issue by sergeisakov** [quantumlib/Cirq#6017]
**Description of the issue**
Raising PauliSum to the third power or greater gives incorrect results. It seems this can be fixed by replacing line 844 in https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/ops/linear_combinations.py with `base *= self`.

**How to reproduce the issue**
```
import cirq

n = 4
o = sum(cirq.Z(cirq.LineQubit(q)) for q in range(n))
assert(o * o * o == o**3)
```

**Cirq version**
1.1.0

**Comment by eliottrosenberg** [comment]
Causes https://github.com/quantumlib/Cirq/issues/6011

**Comment by mustapha-saad-codeStar** [comment]
Hi @tanujkhattar, I would love to work on this issue. Please, what are the deliverables?

**Comment by tanujkhattar** [comment]
@codeStar Thanks for expressing your interest! This issue is already taken, so I'd recommend you to look at https://github.com/quantumlib/Cirq/issues/5999 instead.

**Comment by mustapha-saad-codeStar** [comment]
No worries man, all good. If the issue is still open, you can tag me and I would be glad to help fix it...

**Comment by pavoljuhas** [comment]
Fixed by #6019
